alacritty/src/event.rs

```diff
@@ -940,6 +941,50 @@ impl<'a, N: Notify + 'a, T: EventListener> input::ActionContext<T> for ActionCon
         self.display.pending_update.dirty = true;
     }
 
+    #[inline]
+    fn start_seeded_search(&mut self, direction: Direction, text: String) {
+        let origin = self.terminal.vi_mode_cursor.point;
+
+        // Start new search.
+        self.clear_selection();
+        self.start_search(direction);
+
+        // Enter initial selection text.
+        for c in text.chars() {
+            if let '$' | '('..='+' | '?' | '['..='^' | '{'..='}' = c {
+                self.search_input('\\');
+            }
```

Comment:
that looks a bit weird to say, could we use  `matches!` instead?

Reply:
How would matches make things simpler? It would just be a macro with essentially the same syntax?

Reply:
![tmp](https://github.com/user-attachments/assets/0f2dccfd-9eb7-4d5a-aed1-a212d556ed3f)

It's both longer and a macro (additional indirection). It doesn't really look any cleaner to me in direct comparison, but let me know if you disagree.

Reply:
I don't think that's unusual at all for Rust. That's generally how you match enums even without PartialEq impl.